enterprise/suppress-user/internal/badgerdb/badgerdb.go

```diff
@@ -333,3 +346,29 @@ func (l blogger) Warningf(fmt string, args ...interface{}) {
 func keyPrefix(workspaceID, userID string) string {
 	return fmt.Sprintf("%s:%s:", workspaceID, userID)
 }
+
+func getMetadataFromBadgerItem(item *badger.Item) (*model.Metadata, error) {
+	itemValue, err := item.ValueCopy(nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not copy item value: %w", err)
+	}
+	var badgerDBValue badgerDBValue
+	if len(itemValue) == 0 {
+		return nil, nil
+	}
+	err = json.Unmarshal(itemValue, &badgerDBValue)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal metadata: %w", err)
+	}
+	createdAt, err := time.Parse(time.RFC3339Nano, badgerDBValue.CreatedAt)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse createdAt: %w", err)
+	}
+	return &model.Metadata{
+		CreatedAt: createdAt,
+	}, nil
+}
+
+type badgerDBValue struct {
+	CreatedAt string
```

Comment:
why not `time.Time` instead of `string`?

Reply:
After having a closer look, why do we even need the `badgerDBValue` struct?

Reply:
I used this struct for changing metadata into bytes and vice versa. I used this because badbgerdb can only save byte array as value.

Reply:
wouldn't a marshalled/unmarshalled `model.Metadata` suffice and take away the need for a new struct and custom data parsing?

Reply:
Yeah, made the change